modules/waardexBidAdapter.js

```diff
@@ -0,0 +1,219 @@
+ * @param {String} bidderRequest.gdprConsent.gdprApplies
+ *
+ * @returns {Object} - {
+ *   ua: string,
+ *   language: string,
+ *   [referer]: string,
+ *   [us_privacy]: string,
+ *   [consent_string]: string,
+ *   [consent_required]: string,
+ *   [coppa]: boolean,
+ * }
+ * */
+function getCommonBidsData(bidderRequest) {
+  const payload = {
+    ua: navigator.userAgent || '',
+    language: navigator.language && navigator.language.indexOf('-') !== -1 ? navigator.language.split('-')[0] : '',
+
+  };
+  if (bidderRequest && bidderRequest.refererInfo) {
+    payload.referer = encodeURIComponent(bidderRequest.refererInfo.referer);
+  }
+  if (bidderRequest && bidderRequest.uspConsent) {
+    payload.us_privacy = bidderRequest.uspConsent;
+  }
+  if (bidderRequest && bidderRequest.gdprConsent) {
+    payload.gdpr_consent = {
+      consent_string: bidderRequest.gdprConsent.consentString,
+      consent_required: bidderRequest.gdprConsent.gdprApplies,
+    }
+  }
+  payload.coppa = !!config.getConfig('coppa');
+
+  return payload;
+}
+
+/**
+ * this function checks either bid response is valid or noе
+ *
+ * @param {Object} bid
+ * @param {string} bid.requestId
+ * @param {number} bid.cpm
+ * @param {string} bid.creativeId
+ * @param {number} bid.ttl
+ * @param {string} bid.currency
+ * @param {number} bid.width
+ * @param {number} bid.height
+ * @param {string} bid.ad
+ *
+ * @returns {boolean}
+ * */
+function isBidValid(bid) {
+  if (!bid.requestId || !bid.cpm || !bid.creativeId || !bid.ttl || !bid.currency) {
+    return false;
+  }
+
+  return Boolean(bid.width && bid.height && bid.ad);
+}
+
+/**
+ * @param {Object} serverBid
+ *
+ * @returns {Object|null}
+ * */
+function createBid(serverBid) {
+  const bid = {
+    requestId: serverBid.id,
+    cpm: serverBid.price,
+    currency: 'USD',
+    width: serverBid.w,
+    height: serverBid.h,
+    creativeId: serverBid.crid,
+    netRevenue: true,
+    ttl: 3000,
+    ad: serverBid.adm,
+    dealId: serverBid.dealid,
+    meta: {
+      cid: serverBid.cid,
+      adomain: serverBid.adomain,
+      mediaType: serverBid.ext.mediaType
+    },
+  };
+
+  return isBidValid(bid) ? bid : null;
+}
+
+export const spec = {
+  code: BIDDER_CODE,
+  supportedMediaTypes: [BANNER],
+
+  isBidRequestValid: (bid) => Boolean(bid.bidId && bid.params && +bid.params.placementId && +bid.params.pubId),
+
+  /**
+   * @param {Object[]} validBidRequests -  array of valid bid requests
+   * @param {Object} bidderRequest - an array of valid bid requests
+   *
+   * */
+  buildRequests(validBidRequests, bidderRequest) {
+    const payload = getCommonBidsData(bidderRequest);
+    payload.bidRequests = buildBidRequests(validBidRequests);
+
+    let pubId = '';
+    if (validBidRequests[0] && validBidRequests[0].params && +validBidRequests[0].params.pubId) {
+      pubId = +validBidRequests[0].params.pubId;
+    }
+    const port = location.protocol === 'https:' ? httpsPort : httpPort;
```

Comment:
Since Prebid 3.0, we only allow secure protocols. So, please remove `http` ports here.

Reply:
done